### DISCUSSION TITLE
How about adding Range+Index to allow range syntax too?

### DISCUSSION THREAD

**Discussion by kzu** [manuelroemer/IsExternalInit#8] [Ideas]
Hey there,
I love your Nullable and IsExternalInit polyfills. I noticed there's another (slightly more involved) one that would be awesome to combine with those two: support for range syntax on NS2 (probably others too?).

The code needed to support that is basically https://github.com/devlooped/avatar/blob/main/src/Avatar.UnitTests/Range.cs (which I copy-pasted from corefx, since that was an earlier, simpler version), which you can see working on that net472 multi-targeting project. I use that same file in other NS2 projects too.

I'm just too lazy to clone the entire approach you already have, so I thought I might just suggest you do it instead, he :). 

And FWIW, I think a meta-package that brought them all would be nice too :).

**Comment by AraHaan** [comment]
@kzu can you check if that Range and Index source will compile for NS1.0? I would like it to work with netstandard1.0 and the TFMs implementing it.

**Comment by kzu** [comment]
After adding `System.ValueTuple` package reference, enable `<Nullable>true</Nullable>` and `<LangVersion>Latest</LangVersion>`, as well as replacing one instance of `Array.Empty<T>` (which does not exist in NS1.0) with `new T[0]`, it indeed does compile. Quite amazing :)

**Reply by manuelroemer** [reply]
Wait, so did adding `System.ValueTuple` provide support for Range and Index? I wouldn't have expected that given the name.

**Reply by AraHaan** [reply]
@manuelroemer that package does not provide Index or Range.

**Reply by kzu** [reply]
No, I mean that the System.ValueTuple is needed for the Index/Range in the linked source to build against NS1.0!
